spec/models/miq_widget_spec.rb

```diff
@@ -320,33 +320,50 @@ def add_dashboard_for_user(db_name, userid, group)
     end
 
     it "skips task creation and records warn message if MiqTask for generating widget content exists and not finished" do
-      MiqTask.create(:name => "Generate Widget: '#{@widget.title}'", :state => "Queued", :status => "Ok", :userid => "system")
+      task = MiqTask.create(:name => "Generate Widget: '#{@widget.title}'", :state => "Queued", :status => "Ok", :userid => "system")
       expect($log).to receive(:warn).with(skip_message(@widget))
-      @widget.queue_generate_content
+
+      task_id = @widget.queue_generate_content
+
+      expect(task_id).to eq(task.id)
     end
 
     it "returns MiqTask id if successful and not records warn message" do
       expect($log).not_to receive(:warn).with(skip_message(@widget))
-      return_value = @widget.queue_generate_content
-      expect(return_value).to equal(MiqTask.where(:name => "Generate Widget: '#{@widget.title}'",
-                                                  :id   => @widget.reload.miq_task_id).first.id)
+
```

Comment:
😉 `and not records warn message`

I can't believe I saw that and @Fryguy didn't fit it.  JK, it can be fixed later since this needs to go back other branches